electron/settingsRenderer.js

```diff
@@ -281,9 +321,39 @@ document.addEventListener("DOMContentLoaded", function(event) {
     let sortMethodSelected = $("#sorting-method-select").val()
     ipcRenderer.send('settingsChanged', {key: "sortMethod", value: sortMethodSelected})
   })
-  $("#resetWinLoss").click((e) => {
-    console.log("resetting win/loss")
-    ipcRenderer.send('settingsChanged', {key: "winLossCounter", value: {win: 0, loss: 0}})
+  $(".reset-button").click((e) => {
+    let deck_id = e.target.getAttribute('data-deckid');
+
+    let message = "Are you sure you want to reset (delete) ";
+    if (deck_id == 'all'){
+      message += 'all win/loss counters?';
+    } else if (deck_id == 'all-decks' ){
+      message += 'all deck win/loss counters?'
+    } else if (deck_id == 'total'){
+      message += 'the total win/loss counter?'
+    } else {
+      message += 'the ' + settingsData.winLossObj[deck_id].name + ' win/loss counter?';
+    }
+    if (!dialog.showMessageBox(remote.getCurrentWindow(), {'buttons': ['Cancel','Ok'],'message':message,})){
+      return;
+    }
+
+    console.log("resetting win/loss");
+    let new_wlc = {};
+
+    if (deck_id == 'all') {
+      new_wlc = {'total':{'win':0,'loss':0}};
+    } else if (deck_id == 'all-decks') {
+      new_wlc.total = settingsData.winLossObj.total;
+    } else if (deck_id == 'total'){
+      new_wlc = settingsData.winLossObj;
+      new_wlc.total = {'win':0,'loss':0};
+    } else {
+      new_wlc = settingsData.winLossObj;
+      new_wlc[deck_id] = undefined;
```

Comment:
Because delete kept returning false and not removing the object. {} keeps the key in the object. This method ensure that the property is removed.

Reply:
That's really strange. Are we using different versions of electron / node (/ chrome?!)?

```
Spencatro@LAPTOP-2K3EQNOB ~/PycharmProjects/mtgatracker/electron (user/shawkins/reddit_feedback)
λ ./node_modules/.bin/electron --version

v1.8.8
 Spencatro@LAPTOP-2K3EQNOB ~/PycharmProjects/mtgatracker/electron (user/shawkins/reddit_feedback)
λ node --version
v8.9.4
```

Reply:
BTW, the 'set to undefined' technique isn't working anymore either. And delete is still broken. So I had to resort to the lowest of the low hacks: iterate and append. Yes I am ashamed. But it does work, unlike the correct ways to do it.

Reply:
I think the delete modifier not working is actually a result of using remote for ipc rather than anything wrong with the code. The docs suggest that delete will fail if a property is non-configurable. 

In the below image, I printed the configurable status of winLossCounter in the main process(console window). It is configurable there. Once we hit the settings renderer(second window), we call remote.getGlobal('winLossCounter'). I printed the configurable status out in the console there and it became false.

![configurable](https://i.imgur.com/pZpfS78.png "configurable")

Also just wanted to say good job to both you and Khoross. I love how the new counter page looks/feels!

Reply:
Aha, mystery solved! Nice find @WizardRubic (and I definitely agree; everything from the fa-icon choice to the layout of the buttons is awesome!)